src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -3660,13 +3661,14 @@ citus_internal_delete_shard_metadata(PG_FUNCTION_ARGS)
 		/* this UDF is not allowed allowed for executing as a separate command */
 		EnsureCoordinatorInitiatedOperation();
 
-		if (!ShardExists(shardId))
+		bool missingOk = false;
+
+		if (!ShardExists(shardId, missingOk))
```

Comment:
with missingOk is false, the error below is never reached. It seems like the intention of ShardExists is to always have missingOk is true, so the argument can probaby be removed.